src/test/Spec.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ViewPatterns #-}
+
+import Test.Tasty
+import Test.Tasty.HUnit as HU
+
+import Echidna.Campaign (Campaign(..), tests, campaign, TestState(..))
+import Echidna.Config (EConfig, defaultConfig, sConf, parseConfig)
+import Echidna.Solidity (loadTesting, quiet)
+import Echidna.Test (SolTest)
+import Echidna.Transaction (Tx, call)
+
+import Control.Lens ((^.), (&), (.~), view)
+import Control.Monad.Reader (runReaderT)
+import Data.Maybe (fromJust)
+import Data.Text (Text)
+import EVM.ABI (AbiValue(..))
+
+main :: IO ()
+main = defaultMain spec
+
+spec :: TestTree
+spec = testGroup "Echidna" [solidityTests]
+
+solidityTests :: TestTree
+solidityTests = testGroup "Solidity-HUnit"
+  [ testCase "Always True" $ testContract' c5 $
+      \c -> do
+        let tr = findtest' c "echidna_true"
+        assertBool "echidna_true somehow did not pass" $ passed tr
+  , testCase "Simple Flags Example" $ testContract' c2 $
+      \c -> do
+        let t1 = findtest' c "echidna_alwaystrue"
+            t2 = findtest' c "echidna_sometimesfalse"
+        assertBool "echidna_alwaystrue did not pass" $ passed t1
+        assertBool "echidna_sometimesfalse unsolved" $ solved t2
+        let sol = solve t2
+        assertBool "solution has length /= 2" $ length sol == 2
+        -- the following check will fail with significantly higher probability
+        -- due to its higher requirements of being met
+        -- let calls@(call1:call2:[]) = view call <$> sol
+        -- assertBool ("solution is not 'set0(0);set1(0);', was '" ++ show calls ++ "'") $
+        --   case call1 of
+        --        Left ("set0", [AbiInt _ 0]) -> True
+        --        _                           -> False
+        --   &&
+        --   case call2 of
+        --        Left ("set1", [AbiInt _ 0]) -> True
+        --        _                           -> False
+  , testCase "Optimal Solve" $ testContract' c3 $
+      \c -> do
+        let tr = findtest' c "echidna_revert"
+        assertBool "echidna_revert unsolved" $ solved tr
+        let sol = solve tr
+        assertBool "solution has length /= 1" $ length sol == 1
+        let sol' = head sol
```

Comment:
I'm worried the partial function here will obfuscate the reason for future failures from people trying to contribute to echidna. Let's instead check for the empty list.

Reply:
Done (@japesinator)